File: day:24/levelOrderBinaryTree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun levelOrderBottom(root: TreeNode?): List<List<Int>> {
        // base case
        if(root == null) return emptyList();
        
        var q:MutableList<TreeNode?> = mutableListOf();
        q.add(root)
        
        var Item:MutableList<Int>;
        var result:MutableList<MutableList<Int>> = mutableListOf();
    
    
        while(q.size > 0) {
            var qq: MutableList<TreeNode?> = mutableListOf();
            var list:MutableList<Int> = mutableListOf();
            for(temp in q) {
                if(temp == null) continue;
                list.add(temp.`val`);
                if(temp.left != null) qq.add(temp.left);
                if(temp.right != null) qq.add(temp.right);
            }
            result.add(0, list);
            q = qq;
        }
        return result;
    }
}

File: day:24/levelOrderBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        #base case
        if(root is None): return [];
    
        q = [root];
        result = [];
    
        while(len(q) > 0):
            qq = [];
            list = [];
            for i in range(len(q)):
                temp = q[i];
                if(temp is None): continue;
                list.append(temp.val);
                if(temp.left): qq.append(temp.left);
                if(temp.right): qq.append(temp.right);
            result.insert(0, list);
            q = qq;
        return result;
